src/Workspaces/Core/Portable/Shared/Extensions/IAsyncEnumerableExtensions.cs

```diff
@@ -20,5 +34,51 @@ await foreach (var value in values.WithCancellation(cancellationToken).Configure
 
             return result.ToImmutable();
         }
+
+        /// <summary>
+        /// Takes an array of <see cref="IAsyncEnumerable{T}"/>s and produces a single resultant <see
+        /// cref="IAsyncEnumerable{T}"/> with all their values merged together.  Absolutely no ordering guarantee is
+        /// provided.  It will be expected that the individual values from distinct enumerables will be interleaved
+        /// together.
+        /// </summary>
+        /// <remarks>This helper is useful when doign parallel processing of work where each job returns an <see
+        /// cref="IAsyncEnumerable{T}"/>, but one final stream is desired as the result.</remarks>
+        public static IAsyncEnumerable<T> MergeAsync<T>(this ImmutableArray<IAsyncEnumerable<T>> streams, CancellationToken cancellationToken)
+        {
+            // Code provided by Stephen Toub, but heavily modified after that.
+
+            var channel = Channel.CreateUnbounded<T>();
+
+            var tasks = new Task[streams.Length];
+            for (var i = 0; i < streams.Length; i++)
+                tasks[i] = Process(streams[i], channel.Writer, cancellationToken);
+
+            // Note: passing CancellationToken.None here is intentional/correct.  We must complete all the channels to
+            // allow reading to complete as well.
+            Task.WhenAll(tasks).ContinueWith(t =>
+            {
+                if (t.IsFaulted)
+                    channel.Writer.Complete(t.Exception);
+                else
+                    channel.Writer.Complete();
```

Comment:
You could simplify this whole body to just the:
```C#
channel.Writer.Complete(t.Exception);
```
line. Task.Exception only returns non-null if the Task is faulted, and Complete(null) means success.

Reply:
> Gets the [AggregateException](https://learn.microsoft.com/en-us/dotnet/api/system.aggregateexception?view=net-7.0) that caused the [Task](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task?view=net-7.0) to end prematurely. If the [Task](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task?view=net-7.0) completed successfully or has not yet thrown any exceptions, this will return null.

Awesome!

Reply:
done.